shell/platform/android/io/flutter/embedding/engine/systemchannels/PlatformViewsChannel.java

```diff
@@ -0,0 +1,303 @@
+            final int id = call.arguments();
+            platformViewsMethodHandler.disposePlatformView(id);
+            result.success(null);
+          } catch (Exception exception) {
+            result.error(
+                "error",
+                exception.getMessage(),
+                null
+            );
+          }
+
+          return;
+        case "resize":
+          try {
+            final PlatformViewResizeConfiguration config = PlatformViewResizeConfiguration.fromMessage((Map<String, Object>) call.arguments());
+            platformViewsMethodHandler.resizePlatformView(config, new SuccessCallback() {
+              @Override
+              public void onSuccess() {
+                result.success(null);
+              }
+
+              @Override
+              public void onFailure() {
+                result.error("error", "Failed to resize view: " + config.id, null);
+              }
+            });
+          } catch (Exception exception) {
+            result.error(
+                "error",
+                exception.getMessage(),
+                null
+            );
+          }
+
+          return;
+        case "touch":
+          try {
+            final TouchInfo touchInfo = TouchInfo.fromMessage((List<Object>) call.arguments());
+            platformViewsMethodHandler.onTouch(touchInfo);
+          } catch (Exception exception) {
+            result.error(
+                "error",
+                exception.getMessage(),
+                null
+            );
+          }
+
+          return;
+        case "setDirection":
+          try {
+            Map<String, Object> args = call.arguments();
+            int id = (int) args.get("id");
+            int direction = (int) args.get("direction");
+            validateDirection(direction);
+
+            platformViewsMethodHandler.setDirection(id, direction);
+          } catch (Exception exception) {
+            result.error(
+                "error",
+                exception.getMessage(),
+                null
+            );
+          }
+
+          return;
+      }
+      result.notImplemented();
+    }
+  };
+
+  public PlatformViewsChannel(@NonNull DartExecutor dartExecutor) {
+    channel = new MethodChannel(dartExecutor, "flutter/platform_views", StandardMethodCodec.INSTANCE);
+    channel.setMethodCallHandler(parsingMethodHandler);
+  }
+
+  // TODO(mattcarroll): javadoc
+  public void setPlatformViewsMethodHandler(@Nullable PlatformViewsMethodHandler platformViewsMethodHandler) {
+    this.platformViewsMethodHandler = platformViewsMethodHandler;
+  }
+
+  // TODO(mattcarroll): javadoc
+  public void overrideDefaultMessageHandler(@NonNull MethodChannel.MethodCallHandler methodCallHandler) {
+    channel.setMethodCallHandler(methodCallHandler);
+  }
+
+  // TODO(mattcarroll): javadoc
+  public void restoreDefaultMessageHandler() {
+    channel.setMethodCallHandler(parsingMethodHandler);
+  }
+
+  public interface PlatformViewsMethodHandler {
```

Comment:
@amirh it doesn't appear to be implemented.  Unfortunately, it's now been so long since this change that I can't remember why.  I'm also guessing your work has evolved a lot since this point, right?  Do you think you could help me get this implementation up to date with your work?